tests/cases/fourslash/quickInforForSucessiveInferencesIsNotAny.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="fourslash.ts" />
+
+////declare function schema<T> (value : T) : {field : T};
+////
+////declare const b: boolean;
+////const obj/*1*/ = schema(b);
+////const actualTypeOfNested/*2*/ = schema(obj);

```

Comment:
Interesting. This is not circular generic type though so this shouldn't run into the circularity print in the first place. So to me it feels like we are missing something else that we can account in ID (eg. instantiations of the type as well?)

Reply:
Right, so we track how many times we visit the same symbol, since we don't know if the symbol is generative or not (ie, does each member produce a type whose member is a new instantiation of the type) and we have tests to that effect. In this case, we do produce a type with a member of the same symbol as the overall type, but it's finite - just 2 levels deep. The explicit `depth` check helps distinguish these cases.

Reply:
I understand why you are tracking depth but what i meant was why not add (symbol.links.instatiations.typeId) to determine if we are recuring into symbol + type combination. Thus we would still not have to print something like `z : { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: { y2: any; }; }; }; }; }; }; }; }; }; }; }`

Reply:
Every `y2` in that type is within a differently id'd object literal type backed by the same (well, 1 of 2, the test is mutual recursion) symbol. No `symbol + type` id pair will ever repair because it generates new `type` ids infinitely as it recurs.